Markup and styling changes for example code

sarajw opened the pull request:
Hi!

So here I've made a point to style the markup inline, so it looks like a proper 90s table.

But, you can pull all the 'style' attributes out, and it looks fine.

It might be worth going one step further and offering two boxes to copy, one for a `<style>` tag to go into the `<head>`, and then the plain markup without styles within another.

But I wanted to just do this first to start the conversation - I don't expect this to be pulled in as-is. I've made a little playground here:
https://codepen.io/sarajw/pen/BaqbQGe

Thanks for being cool with me contributing :)

Edit: Just to say I've not pulled this all locally to my machine and run it live, so I've edited code without checking it works on the site itself - so it may bring about bugs and apologies if so!

sarajw committed:
Update site.js with new links markup

Changed to div elements and styled to look like old table borders

sarajw committed:
Update webring.pug

sarajw committed:
Update webring.pug

sarajw committed:
Update site.js

sarajw commented on src/site/api/view/webring/webring.pug:
```diff
@@ -31,15 +31,13 @@ div.content-block
-		| 	</tr>
-		| </table>
+	textarea#example-code(rows=6)
+		| <divstyle="width: fit-content; border: 2px outset; text-align:center">
```
Oh no. Already missed a space between div and style

ajxs commented:
This is really awesome work! Maybe I could build on top of your example on codepen, and make a visual example gallery for users? I'll give this a try tomorrow and see if I can make something usable.

sarajw commented:
Hey thanks! And yes of course, do whatever you want with it :)

sarajw commented:
Hello Anthony!

I may have found a bug - or just stumbled into something that isn't quite allowed?

On the CS.Sjoy.lol ring, I suspect that sites with URLs that include an anchor on the end (like domain.com/about/#webring-id) seem to mess up the order of the sites in the prev/next links. Does that sound likely?

Thanks for a really useful service!


On Thu, 25 May 2023, at 15:26, Anthony wrote:
> 
> 
> This is really awesome work! Maybe I could build on top of your example on codepen, and make a visual example gallery for users? I'll give this a try tomorrow and see if I can make something usable.
> 
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/webri-ng/webri.ng/pull/2#issuecomment-1562907657>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AVDPFIVYSKRSMOT77PXBQW3XH5MYPANCNFSM6AAAAAAYOUS6YI>.
> You are receiving this because you authored the thread.Message ID: ***@***.***>
>

ajxs commented:
No problem! It's my pleasure.
I took a look at the webring you mentioned. I think the problem could be that the final site has the URL
`https://domain.com/about/#webrings`, but the markup has `/next?via=https://domain.com/about/#webring`.
The `via=` 'referral URL' code works by querying the list of sites for one matching the provided URL.
When I tried the 'previous' link with the 's' added to the end, I ended up at the right place.

ajxs commented:
As for the example markup, I've been a bit busy this weekend, but I'll have something finished this week! I really like your examples, and I want to try working that into the site. There's lots of room for improvement in this area!

sarajw commented:
Ahaaaaaa thank you! Silly mistake, was clearly too tired when messing with my links last night. Thank you! Sorry for bothering you with it.

On Sun, 28 May 2023, at 10:30, Anthony wrote:
> 
> 
> 
> No problem! It's my pleasure.
> I took a look at the webring you mentioned. I think the problem could be that the final site has the URL
> `https://domain.com/about/#webrings`, but the markup has `/next?via=https://domain.com/about/#webring`.
> The `via=` 'referral URL' code works by querying the list of sites for one matching the provided URL.
> When I tried the 'previous' link with the 's' added to the end, I ended up at the right place.
> 
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/webri-ng/webri.ng/pull/2#issuecomment-1565995402>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AVDPFIUJ2FL3XK575FEWFO3XIMEJLANCNFSM6AAAAAAYOUS6YI>.
> You are receiving this because you authored the thread.Message ID: ***@***.***>
>

sarajw commented:
Awesome! I'm very into CSS as you can probably guess, so if you have ideas for this but no time to style them up, I'm v. happy to contribute :)

On Sun, 28 May 2023, at 10:31, Anthony wrote:
> 
> 
> As for the example markup, I've been a bit busy this weekend, but I'll have something finished this week! I really like your examples, and I want to try working that into the site. There's lots of room for improvement in this area!
> 
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/webri-ng/webri.ng/pull/2#issuecomment-1565995644>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AVDPFIQJXH7WSRDDGY7EFTTXIMENRANCNFSM6AAAAAAYOUS6YI>.
> You are receiving this because you authored the thread.Message ID: ***@***.***>
>

sarajw commented:
Mmm, there's still something funny going on. https://sarajoy.dev/#webring has got the links right as far as I can tell, but both prev and next links go to the first site in the webring list, and carries the #webring along with it 🤔

On Sun, 28 May 2023, at 10:40, Sara Joy Wallén wrote:
> Ahaaaaaa thank you! Silly mistake, was clearly too tired when messing with my links last night. Thank you! Sorry for bothering you with it.
> 
> On Sun, 28 May 2023, at 10:30, Anthony wrote:
>> 
>> 
>> 
>> 
>> No problem! It's my pleasure.
>> I took a look at the webring you mentioned. I think the problem could be that the final site has the URL
>> `https://domain.com/about/#webrings`, but the markup has `/next?via=https://domain.com/about/#webring`.
>> The `via=` 'referral URL' code works by querying the list of sites for one matching the provided URL.
>> When I tried the 'previous' link with the 's' added to the end, I ended up at the right place.
>> 
>> 
>> 
>> 
>> —
>> Reply to this email directly, view it on GitHub <https://github.com/webri-ng/webri.ng/pull/2#issuecomment-1565995402>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AVDPFIUJ2FL3XK575FEWFO3XIMEJLANCNFSM6AAAAAAYOUS6YI>.
>> You are receiving this because you authored the thread.Message ID: ***@***.***>
>> 
>>

ajxs commented:
> Mmm, there's still something funny going on. https://sarajoy.dev/#webring has got the links right as far as I can tell, but both prev and next links go to the first site in the webring list, and carries the #webring along with it

I checked, and you're right. Sorry, that's my mistake.
I think I've figured out why this is happening. The server framework webri.ng uses interprets the trailing 'anchor' in the URL as not being part of the 'via' query parameter. The solution is to 'encode' this URL component on the front-end. I've changed the markup generation so that it automatically encodes the site URLs in a way that the server framework interprets them correctly. It doesn't look as nice, but it looks like it might resolve the issue. When you get a chance, can you try updating the links in your site and see if this helps?

sarajw commented:
Updated, and the ring now completes a full circuit, woohoo!

On Mon, 29 May 2023, at 07:57, Anthony wrote:
> 
> 
>> 
>> Mmm, there's still something funny going on. https://sarajoy.dev/#webring has got the links right as far as I can tell, but both prev and next links go to the first site in the webring list, and carries the #webring along with it thinking
>> … <https://app.fastmail.com/mail/Inbox/compose?u=91494013&theme=dark#>
>> On Sun, 28 May 2023, at 10:40, Sara Joy Wallén wrote: Ahaaaaaa thank you! Silly mistake, was clearly too tired when messing with my links last night. Thank you! Sorry for bothering you with it. On Sun, 28 May 2023, at 10:30, Anthony wrote: > > > > > No problem! It's my pleasure. > I took a look at the webring you mentioned. I think the problem could be that the final site has the URL > `[https://domain.com/about/#webrings`](https://domain.com/about/#webrings%60), but the markup has `/next?via=[https://domain.com/about/#webring`](https://domain.com/about/#webring%60). > The `via=` 'referral URL' code works by querying the list of sites for one matching the provided URL. > When I tried the 'previous' link with the 's' added to the end, I ended up at the right place. > > > > > — > Reply to this email directly, view it on GitHub <#2 (comment) <https://github.com/webri-ng/webri.ng/pull/2#issuecomment-1565995402>>, or unsubscribe https://github.com/notifications/unsubscribe-auth/AVDPFIUJ2FL3XK575FEWFO3XIMEJLANCNFSM6AAAAAAYOUS6YI. > You are receiving this because you authored the thread.Message ID: **@**.***> > >
>> 
> 
> I checked, and you're right. Sorry, that's my mistake.
> I think I've figured out why this is happening. The server framework webri.ng uses interprets the trailing 'anchor' in the URL as not being part of the 'via' query parameter. The solution is to 'encode' this URL component on the front-end. I've changed the markup generation so that it automatically encodes the site URLs in a way that the server framework interprets them correctly. It doesn't look as nice, but it looks like it might resolve the issue. When you get a chance, can you try updating the links in your site and see if this helps?
> 
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/webri-ng/webri.ng/pull/2#issuecomment-1566592706>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AVDPFIXTEPOM6XITFIE5TT3XIQ3FDANCNFSM6AAAAAAYOUS6YI>.
> You are receiving this because you authored the thread.Message ID: ***@***.***>
>

ajxs commented:
I've merged these changes in now, and deployed them to production.
Thank you so much for your help! I'm very sorry about how long it took me to get this into production. It's been a very busy time for me. I thought I'd get some other changes merged in ahead of this one, but that was taking too long. I'm going to continue to look into improving the markup generation. 
If there's any other improvements you think could be made, feel free to get in touch, or just raise a PR directly!
Thanks again!

sarajw commented:
No apologies! If it's a hobby then whenever, do it when you can.

You're so welcome :)

On Sat, 10 Jun 2023, at 09:18, Anthony wrote:
> 
> 
> 
> I've merged these changes in now, and deployed them to production.
> Thank you so much for your help! I'm very sorry about how long it took me to get this into production. It's been a very busy time for me. I thought I'd get some other changes merged in ahead of this one, but that was taking too long. I'm going to continue to look into improving the markup generation.
> If there's any other improvements you think could be made, feel free to get in touch, or just raise a PR directly!
> Thanks again!
> 
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/webri-ng/webri.ng/pull/2#issuecomment-1585530791>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AVDPFIVNWVWEKYCGUQXVT2LXKQNU5ANCNFSM6AAAAAAYOUS6YI>.
> You are receiving this because you authored the thread.Message ID: ***@***.***>
>

sarajw commented:
Hiya!

Random thought that came to me, though not really important as I can always just hand code it:
Do you already, or feel like supplying the list of sites on a webring, accessible by API or script or similar?

Cheers! My ring is slowly growing and my social circles are pretty into webrings as a concept :)

Sara

On Sat, 10 Jun 2023, at 09:18, Anthony wrote:
> 
> 
> 
> I've merged these changes in now, and deployed them to production.
> Thank you so much for your help! I'm very sorry about how long it took me to get this into production. It's been a very busy time for me. I thought I'd get some other changes merged in ahead of this one, but that was taking too long. I'm going to continue to look into improving the markup generation.
> If there's any other improvements you think could be made, feel free to get in touch, or just raise a PR directly!
> Thanks again!
> 
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/webri-ng/webri.ng/pull/2#issuecomment-1585530791>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AVDPFIVNWVWEKYCGUQXVT2LXKQNU5ANCNFSM6AAAAAAYOUS6YI>.
> You are receiving this because you authored the thread.Message ID: ***@***.***>
>

ajxs commented:
Sure! There isn't currently an API for this, but it would be very easy for me to make one. I'll do that this week.
What kind of information do you think would be useful for a webring API in your case? My first thought is to create an API endpoint that returns all of the public information about a webring. Such as the name, URL, and a list of its associated sites.
